### PR TITLE
Page object pattern practice task

### PR DESCRIPTION
**Pull request by NORDBBFG**
For this practical task, you're required to use the result of the previous module. You should proceed to work with Selenium WebDriver along with your chosen unit test framework, either NUnit, MSTest or xUnit. Your .NET project type should be a Class Library.
Your solution should consist of a single class containing several test methods. It would be best if you utilized the page object pattern, but kindly don't use the PageFactory extension. Also, reflect on the appropriate passing of the IWebDriver object to make it user-friendly. Consider using the Actions API to execute a scroll wheel action in your code at least once. Additionally, incorporate an option to run tests in headless mode, which should be the second option for test execution.
Tasks https://github.com/NORDBBFG/Selenium_pract_task_epam_repo/pull/1 and https://github.com/NORDBBFG/Selenium_pract_task_epam_repo/pull/2:
Refactor tasks created for the previous module to use the PageObject pattern to abstract any page information away from the actual.
Precondition: Execute test cases manually before creating automated tests. Make all tests parameterized to practice the Data-Driven approach.

Test case https://github.com/NORDBBFG/Selenium_pract_task_epam_repo/pull/3. Validate file download function works as expected:
Create a Chrome instance.
Navigate to https://www.epam.com/.
Select “About” from the top menu.
Scroll down to the “EPAM at a Glance” section.
Click on the “Download” button.
Wait till the file is downloaded.
Validate that file “[EPAM_Systems_Company_Overview.pdf](https://www.epam.com/content/dam/epam/free_library/EPAM_Systems_Company_Overview.pdf)” downloaded (use the name of the file as a parameter)
Close the browser.

Test case #4. Validate title of the article matches with title in the carousel:
Create a Chrome instance.
Navigate to https://www.epam.com/.
Select “Insights” from the top menu.
Swipe a carousel twice.
Note the name of the article.
Click on the “Read More” button.
Validate that the name of the article matches with the noted above.
Close the browser.

### CODE REVIEW

File: Selenium_pract_task/POM/EpamMainPage/EpamAboutPage/EpamAboutPageContext.cs [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+using Selenium_pract_task.Entities.AbstractEntities;
+using static Selenium_pract_task.Constants.Constants.IOConstants;
+using static Selenium_pract_task.FileHelper.FileHelper;
+using NUnit.Framework;
+
+namespace Selenium_pract_task.POM.EpamMainPage.EpamAboutPage
+{
+    public class EpamAboutPageContext : AbstractPageContext
+    {
+        EpamAboutPage epamAboutPage;
```
Comment (origami1111) [inline Selenium_pract_task/POM/EpamMainPage/EpamAboutPage/EpamAboutPageContext.cs]: Make private explicitly

File: Selenium_pract_task/POM/EpamMainPage/EpamAboutPage/EpamAboutPageContext.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+        private EpamMainPage epamMainPage;
```
Comment (origami1111) [inline Selenium_pract_task/POM/EpamMainPage/EpamAboutPage/EpamAboutPageContext.cs]: Delete unused field

File: Selenium_pract_task/POM/EpamMainPage/EpamInsightsPage/EpamContinuumPage/EpamContinuumPage.cs
Comment (origami1111) [inline Selenium_pract_task/POM/EpamMainPage/EpamInsightsPage/EpamContinuumPage/EpamContinuumPage.cs:1]: Continuum is not a page, it is an element. Create it like a separate element, inside of it make any methods if necessary (like click, swipe...). And introduce this ContinuumElement inside of InsightsPage. 
P.S.: See UI Element pattern or something like that (you can see at JDI framework that was created by EPAM - https://jdi.epam.com/).

File: Selenium_pract_task/Tets_Cases/Validate_Title_Of_The_Article_Matches_With_Title_In_The_Carousel.cs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    {
+        private string activeSliderText;
+
+        [Test]
+        public void ValidateArticleMaches()
+        {
+            var epamMainPageContext = new EpamMainPageContext(driver);
+            epamMainPageContext.VerifyCookiesHandler()
+                .ClickOnLinkInsightsLink()
+                .ClickButtonNextSlideContinuum()
```
Comment (origami1111) [inline Selenium_pract_task/Tets_Cases/Validate_Title_Of_The_Article_Matches_With_Title_In_The_Carousel.cs]: Make ClickOnLinkInsightsLink as parametrized with a number to click on this button

**Review by origami1111 — CHANGES_REQUESTED** [review]
See comments

File: Selenium_pract_task/Tets_Cases/Validate_File_Download_Function-Works-As_Expected.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+using Selenium_pract_task.Entities.AbstractEntities;
+using Selenium_pract_task.POM.EpamMainPage;
+using static Selenium_pract_task.Constants.Constants.IOConstants;
+
+namespace Selenium_pract_task.Tets_Cases
+{
+    [TestFixture]
+    public class Validate_File_Download_Function_Works_As_Expected : BaseTest
+    {
+        private string downloadedFileName = "EPAM_Corporate_Overview_Q4_EOY.pdf";
```
Comment (origami1111) [inline Selenium_pract_task/Tets_Cases/Validate_File_Download_Function-Works-As_Expected.cs]: According to requirements: Validate that file “[EPAM_Systems_Company_Overview.pdf](https://www.epam.com/content/dam/epam/free_library/EPAM_Systems_Company_Overview.pdf)” downloaded (**use the name of the file as a parameter**)

File: Selenium_pract_task/POM/EpamMainPage/EpamInsightsPage/EpamContinuumPage/EpamContinuumPage.cs
Comment (NORDBBFG) [reply Selenium_pract_task/POM/EpamMainPage/EpamInsightsPage/EpamContinuumPage/EpamContinuumPage.cs:1]: page element

**Pull request closed without merging**